.github/workflows/doc-publish-pr.yml

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2020 Linaro Limited.
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+name: Documentation Publish (Pull Request)
```

Comment:
That was sort of the initial implementation, but it was suggested in the infrastructure team meeting that we should create a completely separate/independent workflow for the PR uploads to prevent any potential credentials mix-ups and the subsequent security issues in the future.